pkg/plugins/resources/helm/utils.go

```diff
@@ -263,6 +263,22 @@ forLoop:
 			// Reset Version to its initial value
 			computedVersion = initialComputedVersion
 			break forLoop
+		case AUTO:
+			origVer, oErr := semver.NewVersion(strings.Trim(resultTarget.Information, "~><=^"))
+			newVer, nErr := semver.NewVersion(strings.Trim(resultTarget.NewInformation, "~><=^"))
+
+			if oErr != nil || nErr != nil {
+				computedVersion = v.IncMinor().String()
+				continue
+			}
+
+			if newVer.Major() > origVer.Major() {
+				computedVersion = v.IncMajor().String()
+			} else if newVer.Minor() > origVer.Minor() {
+				computedVersion = v.IncMinor().String()
+			} else if newVer.Patch() > origVer.Patch() {
+				computedVersion = v.IncPatch().String()
```

Comment:
What would you think to also catch downgrade scenarios and prerelease/metadata change? 

```
  if newVer.Minor() > origVer.Minor() || newVer.Minor() < origVer.Minor()
``` 

And at the end of the condition, a catchall for situation like prelease or metadata change

else {
  computedVersion = v.IncMinor().String()
}

Reply:
Hey @olblak! I have attempted to address both of these comments in my most recent commits. I did take the liberty of shortening the `if a > b || a < b` to simply `if a != b`, but other than that it's the same.

I added a new field in the `values.yaml` file in the test chart to allow version downgrades without changing any existing tests. I hope that's okay.

Reply:
Indeed way easier to read